### DISCUSSION TITLE
Sometimes can't fill rectangle

### DISCUSSION THREAD

**Discussion by vrmvgf** [SixLabors/ImageSharp.Drawing#198] [Q&A] [answered]
I draw and fill two rectangles, but the one drawn with variables can't be filled. Is there something wrong?
![bar1](https://user-images.githubusercontent.com/95868784/153019056-fd0d7f52-1459-4e61-bf79-e97d5f949d1d.jpg)

        public void button1_Click(object sender, EventArgs e)
        {
            var g = new SixLabors.ImageSharp.Image<Rgba32>(100,100);
            g.Mutate(x => x.Fill(Color.White));
            int IH2 = 50; int IB=60; int IH1 =30;
            float SCALE = 1.0F; 
            float HS = SCALE * IH2; float B1S = SCALE * IB; float H1S = SCALE * IH1;
            float YSHFT = 50.0F + HS / 2; float XSHFT = 50.0F - B1S / 2;
            IBrush bru1 = Brushes.Percent20(Color.Blue);

            g.Mutate(x => x.Draw(Color.Blue, 2, RectangleF.FromLTRB(XSHFT, YSHFT, XSHFT + B1S, YSHFT - H1S))
            .Fill(bru1, RectangleF.FromLTRB(XSHFT, YSHFT, XSHFT + B1S, YSHFT - H1S)));
            //Can't fill

            g.Mutate(x => x.Draw(Color.Blue, 2, RectangleF.FromLTRB(10, 10, 70, 30))
            .Fill(bru1, RectangleF.FromLTRB(10, 10, 70, 30)));
            //Can fill

            g.Save("bar1.jpg");
            pictureBox1.Image = System.Drawing.Image.FromFile("bar1.jpg");
        }

**Comment by JimBobSquarePants** [comment, marked as answer]
Your rectangle has a negative height. 

![image](https://user-images.githubusercontent.com/385879/153033107-1db0b211-ad9e-435a-a5d0-fa647cbb7c9f.png)

**Reply by vrmvgf** [reply]
Thank you!

**Reply by tocsoft** [reply]
I would say this is probably still a bug despite it being a negative height. if we can outline it then we should be able to fill it.... deciding on the exact behaviour now would be the question. 

Basically should we ignore the fact the hight was negative normalise the region and fill as though it was well formed and positive in the first place, or should the negative texture/brush be applied in some reversed direction (mostly important for image brushes but would probably also effect gradient brushes too.)

**Reply by JimBobSquarePants** [reply]
How were we able to outline it as a rectangle with negative height? Surely that is the bug?

**Reply by antonfirsov** [reply]
Shouldn't `FromLTRB` throw when `Width` or `Height` is negative?

**Reply by JimBobSquarePants** [reply]
That screenshot is actually `System.Drawing` `RectangleF`. I copied their implementation.
